extra_streamlit_components/CookieManager/frontend/src/CookieManager.jsx

```diff
@@ -25,49 +29,48 @@ const CookieManager = (props: ComponentProps) => {
     return value
   }
 
-  const deleteCookie = (cookie) => {
-    cookies.remove(cookie, { path: "/", samesite: "strict" })
-    return true
-  }
-
   const getAllCookies = () => {
     return cookies.getAll()
   }
 
-  const { args } = props
+  const deleteCookie = (cookie) => { 
+    cookies.remove(cookie)
+    return true
+  }
 
   const method = args["method"]
   const cookie = args["cookie"]
-  const value = args["value"]
+  const val = args["val"]  
   const options = args["options"]
+  
+  
+  useMemo(() => {
+    let output = null
+    
+    switch(method) {
+      case "set":
+        output = set(cookie, val, options)
+        break
+      case "get":
+        output = getCookie(cookie) 
+        Streamlit.setComponentValue(output)
+        Streamlit.setComponentReady()
+        break
+      case "getAll":
+        output = getAllCookies()
+        Streamlit.setComponentValue(output)
+        Streamlit.setComponentReady() 
+        break 
+      case "delete":
+        output = deleteCookie(cookie)
+      default:
+          break
```

Comment:
I had initially thought of not returning output from `set` and `delete` (i.e., removing Streamlit.setComponentValue ) because it will prevent extra rerun of python code in serverside. But, then the server wouldn't know if/when the cookie has been set on the client side. (In my workflow for oauth login, i had to set/delete the cookie and rerun the python page). Returning the output would cause the server to rerun the page only after cookie has been set or deleted.

If output wasn't returned and if I manually run `st.rerun()` immediately after calling cookiemanager.delete() then the cookie wouldn't have been set. I would have to wait few seconds for the server to communicate to client and then only rerun(). So having the rerun initiated by client was better than setting arbitrary sleep seconds.

Would be better to add extra parameter in set and delete for user to choose if they want to return value from client or not.

Reply:
Hey, yeah that was the rationale really, trying to reduce the number of re-runs in the client. But you are right, would be better to five users an option. Will implement it.